tests/handlers/records-delete.spec.ts

```diff
@@ -282,6 +288,229 @@ export function testRecordsDeleteHandler(): void {
         expect(aliceQueryWriteAfterAliceRewriteReply.entries![0].encodedData).to.equal(encodedData);
       });
 
+      describe('protocol based deletes', () => {
```

Comment:
General feed back on new tests:

The test suite seems to lack "negative" tests cases which IMO is equally important for protocol-authorization paths (minus the last else-catch test which doesn't go into the protocol-authorization path): 

ie. if your are not author/recipient, can't delete. Or if you don't belong in $global/$contextRole, can't delete.

While I agree the negative cases "should" work just like other Records methods, since delete has its own implementation... consider rolling the negative cases into the tests added. Let me know if I misread the test or if you think this is an overkill.

Reply:
Updated each test to add a failed attempt by `carol` to delete a record.

Reply:
Kinda hoped the negative role test will involve Carol invoking a role somewhere, but I'll leave that to your discretion!